### PR TITLE
Full-Stack Capstone - Booking an Appointment Application.

### PR DESCRIPTION
**Pull request by tan12082001**
# Final group capstone - Book an Appointment

### Team ( 4 memebers ):

* @Simpleshaikh1 
* @tan12082001 
* @Wineshuga 
* @hetrox8 

### Project Information:

* [Book an Appointment application Business Requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md).
* [Project Progress Kanban Board](https://github.com/users/tan12082001/projects/4/views/1)

`NOTE: Working on two separate Repositories.`

### Instructions for Running the local version:

```
- bundle install
- secret_base_key generation
* rm credentials.yml.enc
* $ EDITOR="mate --wait" bin/rails credentials:edit

- configure database
* run `rails db:create`
* run `rails db:migrate`

- Rspec
* bundle exec rails db:schema:load RAILS_ENV=test
* rspec spec
```

#### Implemented Features:
* Created user authentication - user login and signup.
* Created a Navigation panel that consists of links to respective pages.
* Created a Dashboard that lists all the cars present in the database (Did not add any sample data).
* Created Item Details page which can be viewed by clicking on the specific item from the dashboard.
* Created a Reserve form that would create reservations for the item (have to click on the `Reserve` button inside the item details page.
* Created Add new car page (to add new cars to the endpoint).
* Created Delete Item list - renders names of all the cars present (not deleted items) to be soft deleted(can be accessible from the database).
* Created API Documentation.
* Applied basic styles for both mobile and desktop versions.

#### Kanban board initial view:
image

#### Kanban board final view:
image

#### After External review:
image

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# See https://docs.docker.com/engine/reference/builder/#dockerignore-file for more about ignoring files.
```
Comment (Meltrust) [inline .dockerignore:1]: - [ ] The mobile view has not been implemented yet.  I understand this is a work in progress or a future update.


_tip:  If you feel you've completed 90% of the features, kindly say in your PR description what features will be left out to give a heads-up to the final evaluator and reviewers._

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!








#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
+<a name="readme-top"></a>
```
[the remark this answers is not in the document]
Comment (tan12082001) [reply README.md:1]: > * [ ]  Please check the frontend review for the comments. 👍

Implemented the changes:

TestCases:
image

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**